homeassistant/bootstrap.py

```diff
@@ -112,18 +112,13 @@ def async_from_config_dict(config: Dict[str, Any],
     if not loader.PREPARED:
         yield from hass.async_add_job(loader.prepare, hass)
 
+    # Make a copy because we are mutating it.
+    config = OrderedDict(config)
+
     # Merge packages
     conf_util.merge_packages_config(
         config, core_config.get(conf_util.CONF_PACKAGES, {}))
 
-    # Make a copy because we are mutating it.
```

Comment:
This was already mutated by `merge_packages` in the past. Loading will ensure values are {} and never None